### PR TITLE
Features/new api

### PR DESCRIPTION
**Pull request by Redm4x**
First iteration of the new web api. This new api is staying in sync with each new block and maintaining a database of deployment/leases. By processing each messages we can get a more accurate "closed height" than by querying the existing apis.

** This is only a first iteration and there is a lot to refactor to make it more precise and optimised. Here's some things that will be improved in the future:

- Better handle overdrawn account (proportional splitting of balance)
- Compute and save revenue data while processing blocks to not have to recompute it during each queries
- Get rid of latestDownloadedHeight.txt & latestDownloadedHeight.txt files (maybe store in leveldb)

### CODE REVIEW

File: newApi/src/temp_testing/fakeData.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Lease } from "@src/db/schema";
```
Comment (mbeauchamp7) [inline newApi/src/temp_testing/fakeData.ts:1]: We could remove this file

File: newApi/src/shared/constants.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const averageBlockTime = 6.174;
```
Comment (mbeauchamp7) [inline newApi/src/shared/constants.ts:1]: Technically we will be able to calculate the average block time in the future

File: newApi/src/db/schema.ts [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+  index: { type: DataTypes.NUMBER, allowNull: false },
+  isInterestingType: { type: DataTypes.BOOLEAN, allowNull: false, defaultValue: false },
+  isProcessed: { type: DataTypes.BOOLEAN, allowNull: false, defaultValue: false }
+}, {
+  tableName: "message",
+  modelName: "message",
+  sequelize
+});
+
+Transaction.hasMany(Message);
```
Comment (mbeauchamp7) [inline newApi/src/db/schema.ts:267]: Could move these to `buildDatabase`

**Review by mbeauchamp7 — APPROVED** [review]

File: newApi/src/akash/nodeAccessor.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import { sleep } from "@src/shared/utils/delay";
+
+const fetch = require("node-fetch");
+
+const apiEndpoints = [
+  //"http://135.181.60.250:1317",
```
[the remark this answers is not in the document]
Comment (Redm4x) [reply newApi/src/akash/nodeAccessor.ts:6]: It was slower and sometime completely down so I removed it for stability

File: newApi/src/akash/dataStore.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  console.time("size");
+  const blocksSize = await getTotalSize("./data/blocks.db");
+  const txsSize = await getTotalSize("./data/txs.db");
+  console.timeEnd("size");
+  return { blocksSize: blocksSize, txsSize: txsSize };
+};
+
+export const deleteCache = async function () {
+  console.log("Deleting cache...");
+  blocksDb.clear();
```
[the remark this answers is not in the document]
Comment (Redm4x) [reply newApi/src/akash/dataStore.ts:24]: Good catch, the `.clear()` need to be awaited

**Pull request merged**